filament/backend/src/vulkan/VulkanTexture.cpp

```diff
@@ -97,12 +97,31 @@ VulkanTexture::VulkanTexture(VkDevice device, VkPhysicalDevice physicalDevice,
         imageInfo.extent.depth = 1;
     }
 
+    // Determine if we can use the transient usage flag combined with lazily allocated memory.
+    const bool useTransientAttachment =
+        // Transient attachment is requested.
+        preferTransientAttachment &&
+        // Lazily allocated memory is available.
+        context.isLazilyAllocatedMemorySupported() &&
+        // Usage consists of attachment flags only.
+        !any(tusage & ~TextureUsage::ALL_ATTACHMENTS) &&
+        // Usage contains at least one attachment flag.
+        any(tusage & TextureUsage::ALL_ATTACHMENTS);
+    mIsTransientAttachment = useTransientAttachment;
+
     // Filament expects blit() to work with any texture, so we almost always set these usage flags.
+    // We do not add the flags for transient attachments, because that would prevent the usage of
+    // the VK_IMAGE_USAGE_TRANSIENT_ATTACHMENT_BIT flag.
     // TODO: investigate performance implications of setting these flags.
-    constexpr VkImageUsageFlags blittable = VK_IMAGE_USAGE_TRANSFER_DST_BIT |
-            VK_IMAGE_USAGE_TRANSFER_SRC_BIT;
+    const VkImageUsageFlags blittable =
+       useTransientAttachment ?
+       0U :
+       VK_IMAGE_USAGE_TRANSFER_DST_BIT | VK_IMAGE_USAGE_TRANSFER_SRC_BIT;
```

Comment:
We should probably revisit why blitable usages are systematically added, regardless of the `BLIT_` flags -- we shouldn't have to do these shenanigans with `blitable`.

Reply:
Totally agree. I think forcing everything to be blittable sliently is not good, because of the potential use cases, like transient, where it gets in the way.
It is a broader change though, unrelated with transient support perse. Should we just add a separate todo?

Reply:
yes, let's add a todo, and maybe you can have a look next? (or we will). you can even file a github issue.

Reply:
I can definitely have a look. I think some usage flags that are silently added might also be getting in the way of protected content.